docs/gitlab_integration.md

```diff
@@ -0,0 +1,132 @@
+# 1. Install/register GitLab Runner
+GitLab Runner is the open source project that is used to run your jobs and
+send the results back to GitLab. You have to
+[install](https://docs.gitlab.com/runner/install/) it on a machine where
+your analysis will be performed.
+
+Once GitLab Runner is installed, you need to register it with GitLab.
+You can use the following command:
+```sh
+sudo gitlab-runner register \
+    --non-interactive \
+    --url "https://mycompany.gitlab.com" \
+    --registration-token "$REGISTRATION_TOKEN" \
+    --description "codechecker" \
+    --executor "shell"
+```
+
+The token can be found in your projects settings under
```

Comment:
I would probably put the token generation part before the registration because it is needed for that and a highlight a bit more that this token is generated on the web ui of gitlab.